### DISCUSSION TITLE
How to Improve Throughput Performance on Gateway-to-Gateway?

### DISCUSSION THREAD

**Discussion by ygm521** [strongswan/strongswan#954] [General]
# Test Conditions
Gateway-to-Gateway，tunnel mode and esp ；
- server
iperf3 -s -p 8000
- client
iperf3 -c serIP -u -p 8000 -t 60 -f m -b50000M -P 8 -l 512
- result
121 Mbits/sec

# Analyze the reasons 
Since the data packets of gateway A and gateway B are transmitted, the hash result of calculating sip+dip is the same, and they are finally processed by the same queue of the network card. Therefore, multi-core encryption and decryption are not used to improve performance.

So I try to refer to the kernel rps algorithm, each esp packet is distributed to different cores in round robin，but The result is not ideal。

# question
Have other suggestions? I think the kernel can run at least half of the 10 Gigabit.

**Comment by ygm521** [comment]
get_rps_cpu  skb_get_hash is changed and also modify some demo in get_rps_cpu

**Comment by tobiasbrunner** [comment]
strongSwan does not handle ESP/IPsec traffic, the Linux kernel does. So you might want to discuss this on the [netdev mailing list](http://vger.kernel.org/vger-lists.html#netdev).

**Comment by ygm521** [comment]
@tobiasbrunner did you have not encountered similar problem? any body use vpp+strongswan?

**Reply by tobiasbrunner** [reply]
Ah, you're using VPP, then contact that project.
